docs/topics/numbers.md

```diff
@@ -0,0 +1,289 @@
+[//]: # (title: Numbers)
+
+## Integer types
+
+Kotlin provides a set of built-in types that represent numbers.  
+For integer numbers, there are four types with different sizes and, hence, value ranges.
+
+| Type	 |Size (bits)| Min value| Max value|
+|--------|-----------|----------|--------- |
+| Byte	 | 8         |-128      |127       |
+| Short	 | 16        |-32768    |32767     |
+| Int	 | 32        |-2,147,483,648 (-2<sup>31</sup>)| 2,147,483,647 (2<sup>31</sup> - 1)|
+| Long	 | 64        |-9,223,372,036,854,775,808 (-2<sup>63</sup>)|9,223,372,036,854,775,807 (2<sup>63</sup> - 1)|
+
+All variables initialized with integer values not exceeding the maximum value of `Int`
+have the inferred type `Int`. If the initial value exceeds this value, then the type is `Long`.
+To specify the `Long` value explicitly, append the suffix `L` to the value.
+
+```kotlin
+val one = 1 // Int
+val threeBillion = 3000000000 // Long
+val oneLong = 1L // Long
+val oneByte: Byte = 1
```

Comment:
This example shows literal type inferred from the context, but the explanatory text does not mention this Kotlin's feature.

Reply:
Agree. Added a separate page for type inference and linked it in the paragraph above this snippet.

Reply:
It turned out the separate page for type inference doesn't fit in the current ToC. 

If we don't dive deep into the details about constraint systems, not so much left to say, so it needs to be grouped with smth. I tried to group it with the _Smart casts_ section (as in the specification), but smart casts are already described in the section about type casts. I tried to group it with our new article about builder inference but builder inference itself cannot be described before describing at least functions.

So I just changed the wording in the paragraph above to emphasize that the type is inferred. 
BTW, the feature itself is shortly mentioned in the _Basics_ section.

Reply:
This is a specific type of inference that is inherent only to integer literals, so I think it's worth to be explained here.

Reply:
[Added a note about widening.](https://github.com/JetBrains/kotlin-web-site/pull/2409/commits/105a4721e029537fd031751c1c89f3bb917b9a9b) Does it work? I can also add a link to the corresponding section in the specification. WDYT?